NewsApp/app/src/main/java/com/example/newsapp/features/auth/register/RegisterFragment.kt

```diff
@@ -4,17 +4,64 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import com.example.newsapp.R
+import com.example.newsapp.data.entities.User
 import com.example.newsapp.databinding.FragmentRegisterBinding
+import com.example.newsapp.di.NewsManager
+import com.example.newsapp.features.auth.AuthViewModel
+import com.example.newsapp.features.auth.MyViewModelFactory
 
 class RegisterFragment : Fragment() {
     private lateinit var binding: FragmentRegisterBinding
 
+    private val viewModel: AuthViewModel by viewModels {
+        MyViewModelFactory(NewsManager.userRepo)
+    }
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
         binding = FragmentRegisterBinding.inflate(layoutInflater)
         return binding.root
     }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.registerBtn.setOnClickListener {
+            doRegister()
+        }
+
+        viewModel.registrationResultLiveData.observe(viewLifecycleOwner) {
+            if (it) {
+                requireActivity().finish()
+                findNavController().navigate(R.id.newsActivity)
+            }
+        }
+    }
+
+    private fun doRegister() {
```

Comment:
we also need to make sure that the phone number and email are in a valid format. maybe use regex?